sunpy/map/mapbase.py

```diff
@@ -173,6 +173,10 @@ def __init_subclass__(cls, **kwargs):
         `is_datasource_for` attribute.
         This is then passed into the Map Factory so we can register them.
         """
+        if cls.__doc__ is None:
+            cls.__doc__ = GenericMap.__doc__
```

Comment:
What is this for?

Reply:
When the subclass has no doctring.

Reply:
Do we have a subclass without a docstring?

Reply:
Yes, there is one, `HMISynopticMap` in sunpy/map/sources/sdo.py

Reply:
I think a docstring should be added to it, and these lines removed.

Reply:
Okay, but can you please suggest what the doctoring should be for this class? I am getting familiar with sunpy, never used before.

Reply:
These lines should be removed because:
- All of our subclasses already have their own docstrings, so there is no situation in our code where `__doc__` is `None`.
- If any subclass had no docstring, Python *already* gives it the docstring for its parent class.  This code is unnecessary because it is already done by Python.

Reply:
@ayshih When I remove that condition, I get this error:
`TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'`
And this is pointing to `HMISynopoticMap` class which has no doctstring.

Reply:
Yep, it looks like maybe sphinx automatically inherits docstring, but python doesn't.

Reply:
I think these lines are fine, as if we don't have a docstring it's better to have the GenericMap one than nothing.

Reply:
> Yep, it looks like maybe sphinx automatically inherits docstring, but python doesn't.

It's actually that Python will read the parent class docstring for `help()`, but the `__doc__` attribute itself is not inherited.



> I think these lines are fine, as if we don't have a docstring it's better to have the GenericMap one than nothing.

I disagree, because we should never have no docstring.  We should mandate that every subclass have a docstring, even if it's one sentence.  Otherwise, you get silly stuff like this (https://sunpy--5122.org.readthedocs.build/en/5122/code_ref/map.html#id3), where `HMISynopticMap` claims to be "A Generic spatially-aware 2D data array"